Add new capability string for public share

sharidas opened the pull request:
The default name for the public share is
Public link. And this needs to be reflected
in the files_sharing capabilites. This change
introduces name defaultPublicLinkShareName for
this.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.

Please set the following labels:

- Set label "3 - To review" for review or "2 - Development" in case the PR still has open tasks
- Set label "backport-request" if backport is needed
- Assignment: assign to self
- Milestone: set the same as the ticket this PR fixes, or "development" by default
- Reviewers: pick at least one
-->

## Description
<!--- Describe your changes in detail -->
Add `defaultPublicLinkShareName` to the files_sharing capabilities. `defaultPublicLinkShareName` will help the users know that by default public link share name is `Public link`. The translation is enabled for `Public link`.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes <issue_link>

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Adding capabilities `defaultPublicLinkShareName => Public link` to the files_sharing capabilities. Added translation for `Public link`.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- Added unit test.
- Verified by executing `curl --silent -u admin:admin 'http://localhost/testing4/ocs/v1.php/cloud/capabilities?format=json' | json_pp > /tmp/test.xml` and the file is here: 
[test.txt](https://github.com/owncloud/core/files/2698244/test.txt)


## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

sharidas committed:
Add new capability string for public share

The default name for the public share is
Public link. And this needs to be reflected
in the files_sharing capabilites. This change
introduces name defaultPublicLinkShareName for
this.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

PVince81 commented on apps/files_sharing/lib/Capabilities.php:
```diff
@@ -92,6 +97,7 @@ public function getCapabilities() {
 				$public['upload'] = $this->config->getAppValue('core', 'shareapi_allow_public_upload', 'yes') === 'yes';
 				$public['multiple'] = true;
 				$public['supports_upload_only'] = true;
+				$public['defaultPublicLinkShareName'] = $this->l10n->t('Public link');
```
I think we can now wire up the JS code to read it from the capabilities.

You can get them from the JS with OC.getCapabilities() https://github.com/owncloud/core/blob/master/core/js/js.js#L312

sharidas commented:
Current state of the PR:

- [x] Updated the JS code to use the capability updated `apps/files_sharing/lib/Capabilities.php`.
- [x] Updated the unit test for JS code too

PVince81 reviewed (approved):
:+1:
